composer/datasets/in_context_learning_evaluation.py

```diff
@@ -524,143 +528,54 @@ class InContextLearningMultipleChoiceTaskDataset(Dataset):
-                    assert isinstance(gold_idx, int)
-                    preamble += f'{query}{continuation_delimiter}{choices[gold_idx]}'
-            encoded_example = {}
-            query, choices, gold_idx = self.samples[sample_idx]['query'], self.samples[sample_idx][
-                'choices'], self.samples[sample_idx]['gold'],
-            if len(preamble) > 0:
-                query = f'{example_delimiter}{query}'
-
-            # rstrip the continuation delimiter, because the prompt ending in a space results in degenerate output
-            continuation_delimiter_stripped = continuation_delimiter.rstrip()
-
-            if self.prefix_space:
-                choices = [(f' {choice}' if not choice.startswith(' ') else choice) for choice in choices]
-            query += continuation_delimiter_stripped
-            encoded_example['preamble'] = self.tokenizer(
-                preamble
-            )  # if the preamble is empty then these will be 0-length lists, unless the tokenizer adds special tokens to empty strings (e.g. OPT tokenizer)
-
-            if self.tokenizer.eos_token_id is not None and len(
-                    encoded_example['preamble']
-                ['input_ids']) > 1 and encoded_example['preamble']['input_ids'][-1] == self.tokenizer.eos_token_id:
-                encoded_example['preamble']['input_ids'] = encoded_example['preamble']['input_ids'][:-1]
-
-            encoded_example['gold_idx'] = gold_idx
-
-            encoded_example['query'] = self.tokenizer(query, add_special_tokens=False)
-            encoded_example['choices'] = [self.tokenizer(choice, add_special_tokens=False) for choice in choices]
-
-            examples.append(encoded_example)
-
-        return examples
-
-    def __getitem__(self, index):
-        return self.encoded_dataset[index]
-
-    def __len__(self):
-        return len(self.encoded_dataset)
+    def tokenize_example(self, prompt_and_fewshot: str, ctxt: str, example: dict):
+        tokenized_example = super().tokenize_example(prompt_and_fewshot, ctxt, example)
+        choices = example['choices']
+        if self.prefix_space:
+            choices = [(f' {choice}' if not choice.startswith(' ') else choice) for choice in choices]
+        tokenized_example['choices'] = [self.tokenizer(choice, add_special_tokens=False) for choice in choices]
+        tokenized_example['gold'] = example['gold']
+        return tokenized_example
 
     def collate_fn(self, data):
-        inputs = []
-        continuation_indices = []
-        gold_idxs = []
-        choice_groupings = []
+        batch = {
+            'input_ids': [],
+            'continuation_indices': [],
+            'mode': 'icl_task',
+            'labels': [],
+            'gold_indices': [],
+            'choice_groupings': [],
+        }
         for data_pair in data:
+            # TODO: this line is sus idgi
+            choice_start_idx = len(batch['continuation_indices'])
```

Comment:
I believe its indicating to the downstream usage the indices in continuation_indices that correspond to the choice for _this_ MC problem. So the choice groupings can say like 0-4 is Q #1, 4-8 is Q#2, etc

Reply:
checks out